### ISSUE TITLE
Spooky action at a distance. The ampersand in a function pointer becomes a `CT_ARITH` if a specific code follows.

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1284]
Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60027-UNI-21506.cpp" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60027-UNI-21506.cpp"
index f9e3349..6f48c02 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60027-UNI-21506.cpp"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60027-UNI-21506.cpp"
@@ -1,13 +1,13 @@
 struct bar
 {
-    void (Namespace::*method)(Class& param);
+    void (Namespace::*method)(Class & param);
 };
 
-void Class::Foo(void (*callback)(const Class& entry))
+void Class::Foo(void (*callback)(const Class&entry))
 {
 }
 
 void foo()
 {
     int a = 1;  // if you comment this out, the bug stops reproducing
 }
```

Debug support files:
---
[60027-UNI-21506.cpp](https://gist.githubusercontent.com/mihaipopescu/136c0cf22c1b56d655b92a69adf7b9fd/raw/f9e3349c025dbc116b282b04ccb95a1dc1d6683d/60027-UNI-21506.cpp)
[60027-UNI-21506.cpp.log](https://gist.githubusercontent.com/mihaipopescu/136c0cf22c1b56d655b92a69adf7b9fd/raw/27bc7edb18500c77a0c4e5eb2dfb31382f2cce8e/60027-UNI-21506.cpp.log)
[60027-UNI-21506.cpp.unc](https://gist.githubusercontent.com/mihaipopescu/136c0cf22c1b56d655b92a69adf7b9fd/raw/055e75c0461355679ecd32d2bd7f92f9ffb3a018/60027-UNI-21506.cpp.unc)
[Uncrustify.Cpp.cfg](https://gist.githubusercontent.com/mihaipopescu/136c0cf22c1b56d655b92a69adf7b9fd/raw/23d62d9dba5caf352fdc1793792e98eed4b81929/Uncrustify.Cpp.cfg)
[UNI-21506.cpp](https://gist.githubusercontent.com/mihaipopescu/136c0cf22c1b56d655b92a69adf7b9fd/raw/4621e2a05b0878b3d3b9e0ad27a0f314b8a25bd9/UNI-21506.cpp)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by satishp87** [comment]
In unity fork this issue is addressed with below PR
https://github.com/Unity-Technologies/uncrustify/pull/34

**Comment by gmaurel** [comment]
Fixed at 78af3bb958fe603ba778fad654af38b520079bc3, on Dec 19 16:03:19 2017
